core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1440,6 +1445,12 @@ private CompiledQuery createTable(final ExecutionModel model, SqlExecutionContex
         }
     }
 
+    private TableStructure createTableStructureFromLikeTableMetadata(SqlExecutionContext executionContext, CreateTableModel model) {
+        try (TableReader rdr = engine.getReader(executionContext.getCairoSecurityContext(), model.getLikeTableName().token)) {
+            return tableStructureAdapterLikeTable.of(model, rdr.getMetadata());
```

Comment:
By the time this method returns, the reader will be closed, so it's incorrect to use its metadata. Instead, you should read everything from the reader metadata and mutate the original model (or introduce a reusable model field in the `SqlCompiler`) as if it was a fully fledged `CREATE TABLE` statement.

Reply:
Hi @puzpuzpuz, If I understand correctly do you mean to say mutate the CreateTableModel like in the snippet below ?  I wanted to clarify about PartitionBy and Timestamp as they are of type ExpressionNode in CreateTableModel , what should be the value of position for these attributes? 

```
    private TableStructure createTableStructureFromLikeTableMetadata(SqlExecutionContext executionContext, CreateTableModel model) throws SqlException {
        try (TableReader rdr = engine.getReader(executionContext.getCairoSecurityContext(), model.getLikeTableName().token)) {
            model.setCommitLag(rdr.getCommitLag());
            model.setMaxUncommittedRows(rdr.getMaxUncommittedRows());
            TableReaderMetadata rdrMetadata = rdr.getMetadata();
            for (int i = 0; i < rdrMetadata.getColumnCount(); i++) {
                model.addColumn(rdrMetadata.getColumnName(i), rdrMetadata.getColumnType(i), rdr.getSymbolMapReader(i).getSymbolCapacity(), rdrMetadata.getColumnHash(i));
            }
            model.setPartitionBy(SqlUtil.nextLiteral(sqlNodePool, PartitionBy.fromInteger(rdr.getPartitionedBy()), 0));
            model.setTimestamp(SqlUtil.nextLiteral(sqlNodePool, rdrMetadata.getColumnName(rdrMetadata.getTimestampIndex()), 0));
            model.setWalEnabled(rdrMetadata.isWalEnabled());
            return tableStructureAdapterLikeTable.of(model, rdr.getMetadata());
        }
    }

```

Reply:
Position stands for the index of the first token (node) char in the original query text. In this case, you could use either 0 or the position of the source table name (the one that is specified in LIKE clause).

Reply:
I pushed changes to mutate the create table model and populated the fields mentioned here https://github.com/questdb/questdb/issues/2565#issuecomment-1265626605. Kindly take another look. Thanks